src/fides/api/graph/traversal.py

```diff
@@ -411,25 +245,12 @@ def edge_ends_with_collection(_edge: Edge) -> bool:
                     [{', '.join([str(tn.address) for tn in running_node_queue.data])}]""",
                 )
 
-        # Remove nodes that have custom request fields, since we don't care if these are reachable or not.
-        # They will be used independently by the Dynamic Email Erasure Connector.
-        # TODO: ideally we'll update the Traversal code to include these "custom request field datasets"
-        # as part of the main graph. This is a targeted workaround for now.
         remaining_node_keys = {
             key
             for key in remaining_node_keys
-            if not self.traversal_node_dict[key].node.collection.custom_request_fields()
+            if not self.should_exclude_node(self.traversal_node_dict[key])
         }
 
-        # If a policy is provided, we will allow collections without relevant
-        # data categories to remain unreachable without raising any exceptions
-        if self.policy:
-            remaining_node_keys = {
-                key
-                for key in remaining_node_keys
-                if self.policy.applies_to(self.traversal_node_dict[key])
-            }
```

Comment:
Now that I had to add a third scenario in which to ignore unreachable nodes, I took a different approach. The `should_exclude_node` will iterate through the list of filters from `node_filters`. This allows us to keep adding filters in the future.